### ISSUE TITLE
 Establish conventions for documenting allowed child elements

### ISSUE THREAD

**Issue by arthurevans** [Polymer/hydrolysis#218]
_From @addyosmani on March 11, 2014 10:39_

Right now, we have conventions in place for documenting an element's public properties but it's been mentioned that real-world elements may need to also include child elements that could be composed and distributed into their parents.

Our current state is that docs of what child elements a component takes is somewhat implicit in `<content select="<selector>">`. This requires looking through the source to understand the valid content selectors that can be included. (We also have another issue that many of our existing elements don't take child elements, so examples of how to handle this problem are somewhat scarce).

Could we investigate setting up conventions for how child elements should be documented? It appears the content model is one of the only things our current API docs doesn't cover as well as it could.

cc @sjmiles 

_Copied from original issue: Polymer/docs#317_

**Comment by arthurevans** [comment]
_From @ebidel on March 11, 2014 15:36_

For starters, we need examples in the creating elements guide that use `<content>` and `<shadow>`. Also elements that reuse other elements.

This may be a dupe of https://github.com/Polymer/docs/issues/184

**Comment by arthurevans** [comment]
_From @jyasskin on March 12, 2014 16:32_

You already have http://www.polymer-project.org/docs/elements/polymer-ui-elements.html#polymer-ui-menu, whose content model isn't documented.

**Comment by arthurevans** [comment]
_From @sjmiles on March 12, 2014 16:44_

Elements that have a non-trivial content model are the exception not the
norm. Most of the time elements can take arbitrary content.

I'm not sure what prompts the assumption that this is not the case and
provokes code diving into `<content select>`.

Having said that, I agree that this sort of code diving is harmful, I'm
only suggesting that better overall docs are the problem, not some specific
missing convention.

For example, the notion that `polymer-ui-menu` has a specialized content
model it's an incorrect assumption.  It can take any sane content.
`polymer-ui-menu-item` and brethren are constructed to do particularly nice
things as children of `polymer-ui-menu` but neither actually requires the
other.

**Comment by arthurevans** [comment]
I don't think it matters if selecting a specific type of content is the exception or not. If I'm using an element someone else wrote, I don't know whether the element even has a <content> tag. Adopting a convention here would set the expectation that this is a documented part of the element's contract. 

Even if 95% of the elements are simply:

    @children No

Or:

    @children Yes 

For the other 5%, it could be something like:

    @children Accepts the following children:
        - core-toolbar, .core-header    Placed in the header area
        - \*    Placed in the main content area.

In cases like polymer-ui-menu, this could help prevent people from making incorrect assumptions (whether from looking at code samples or code spelunking).

**Comment by rictic** [comment]
+1 to this.
